### PR TITLE
update ruby infra docker images

### PR DESCRIPTION
**Pull request by JoeNunnelley**
- move to new rubygems and bundler versions
- add rails5deps images
- lock ruby version < 2.3 to static bundler version

### CODE REVIEW

File: ruby/2.5.3/Dockerfile [hunk 1]
```diff
@@ -11,8 +11,8 @@ RUN mkdir -p /usr/local/etc \
 ENV RUBY_MAJOR 2.5
 ENV RUBY_VERSION 2.5.3
 ENV RUBY_DOWNLOAD_SHA256 1cc9d0359a8ea35fc6111ec830d12e60168f3b9b305a3c2578357d360fcf306f
-ENV RUBYGEMS_VERSION 2.7.6
-ENV BUNDLER_VERSION 1.17.1
+ENV RUBYGEMS_VERSION 3.0.1
```
Comment (rantler) [inline ruby/2.5.3/Dockerfile]: Should this also be `3.0.2` as we're using in Ruby 2.5.0?

Comment (JoeNunnelley) [reply ruby/2.5.3/Dockerfile]: hm must have fat-fingered that one. thanks.

File: ruby/2.3/Dockerfile [hunk 2]
```diff
@@ -9,6 +9,8 @@ RUN DEBIAN_FRONTEND=noninteractive apt-get -y update && \
   DEBIAN_FRONTEND=noninteractive apt-get purge -y --auto-remove software-properties-common && \
   rm -rf /var/lib/apt/lists/*
 
+RUN gem update --system
```
Comment (courtneyspurgeon) [inline ruby/2.3/Dockerfile]: let's do `gem update --system 3.0.2`, since anything still using this will be old.

File: runit-ruby/2.3/Dockerfile [hunk 3]
```diff
@@ -9,9 +9,11 @@ RUN DEBIAN_FRONTEND=noninteractive apt-get -y update && \
   DEBIAN_FRONTEND=noninteractive apt-get purge -y --auto-remove software-properties-common && \
   rm -rf /var/lib/apt/lists/*
 
+RUN gem update --system
```
Comment (courtneyspurgeon) [inline runit-ruby/2.3/Dockerfile]: Can we pin the rubygems and bundler version in this one too?

**Comment by znep** [comment]
Why are we adding rails5deps?  I thought we were moving away from that.

File: ruby/2.5.3/Dockerfile (hunk 1 above)
Comment (JoeNunnelley) [reply ruby/2.5.3/Dockerfile]: fixed

File: ruby/2.3/Dockerfile (hunk 2 above)
Comment (JoeNunnelley) [reply ruby/2.3/Dockerfile]: updated

File: runit-ruby/2.3/Dockerfile (hunk 3 above)
Comment (JoeNunnelley) [reply runit-ruby/2.3/Dockerfile]: updated

**Comment by courtneyspurgeon** [comment]
🚢 With squash & merge plz.

**Pull request merged**